web/html/src/manager/state/recurring-states.js

```diff
@@ -0,0 +1,236 @@
+            this.getScheduleDetails(id, action);
+        } else if (!action) {
+            this.getRecurringScheduleList();
+        } else {
+            this.setState({action: action});
+        }
+        this.clearMessages();
+    }
+
+    isFiltered = () => {
+        return !!this.state.minionIds;
+    };
+
+    getRecurringScheduleList = () => {
+        // todo create different endpoints for each use case ("/rhn/manager/api/recurringactions/group/id")
+        const entityParams = inferEntityParams();
+        const endpoint = "/rhn/manager/api/recurringactions" + entityParams;
+        return Network.get(endpoint, "application/json").promise
+            .then(schedules => {
+                this.setState({
+                    action: undefined,
+                    selected: undefined,
+                    schedules: schedules.data
+                });
+            }).catch(this.handleResponseError);
+    };
+
+    getScheduleDetails(row, action) {
+        this.setState({selected: row, action: action});
+    }
+
+    handleDetailsAction(row) {
+        this.getScheduleDetails(row, "details");
+        history.pushState(null, null, "#/details/" + row.recurringActionId);
+    }
+
+    handleEditAction(row) {
+        this.getScheduleDetails(row, "edit");
+        history.pushState(null, null, "#/edit/" + row.recurringActionId);
+    }
+
+    toggleActive(schedule) {
+        Object.assign(schedule, {
+            active: !(schedule.active)
+        });
+        this.updateSchedule(schedule);
+    }
+
+    skipNext(item) {
+        /* Write implementation to skip next run */
+        this.handleForwardAction();
+    }
+
+    updateSchedule(schedule) {
+        return Network.post(
+            "/rhn/manager/api/recurringactions/save",
+            JSON.stringify(schedule),
+            "application/json"
+        ).promise.then((data) => {
+            // HACK: propagate the errors from messages to the UI
+            let newMsgs = [];
+            if (data.messages === undefined || data.messages.length === 0) { // no errors from the server
+                newMsgs = MessagesUtils.info(<span>{t("Schedule successully created.")}</span>);
+            } else {
+                const decorator = data.success ? MessagesUtils.info : MessagesUtils.error;
+                newMsgs = decorator.apply(null, data.messages);
+            }
+
+            const msgs = this.state.messages.concat(newMsgs);
+
+            while (msgs.length > messagesCounterLimit) {
+                msgs.shift();
+            }
+
+            this.onMessageChanged(msgs);
+            this.setState({
+                messages: msgs
+            });
+
+            if (data.success) {
+                this.handleForwardAction();
+            } else {
+                this.handleResponseError();
+            }
+
+        }).catch(this.handleResponseError);
+    }
+
+    deleteSchedule(item) {
+        if (!item) return false;
```

Comment:
Why do we need this check?

I also believe the checkstyle would complain about it. It should be:
`if (!item) {
   return false;
}`